packages/amplify_datastore/example/lib/main.dart

```diff
@@ -158,26 +167,69 @@ class _MyAppState extends State<MyApp> {
     });
   }
 
+  savePost() async {
+    try {
+      Post post = Post(
+        title: 'Created Successfully',
+        rating: 15,
+      );
+      await Amplify.DataStore.save(post, when: Post.RATING.ge(10));
+      Post newPost = post.copyWith(id: post.id, title: 'Updated Successfully');
+      await Amplify.DataStore.save(newPost, when: Post.RATING.ge(10));
+    } catch (e) {
+      print(e);
+    }
+  }
+
+  deletePost() async {
+    try {
+      await Amplify.DataStore.delete(
+          // replace 'id' for testing as needed
+          Post(id: 'f6c4be2d-9b1b-496e-b3bf-78035f0e6191', title: 'Title'));
+    } catch (e) {
+      print(e);
+    }
+  }
+
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
       home: Scaffold(
-          appBar: AppBar(
-            title: const Text('Plugin example app'),
+        appBar: AppBar(
+          title: Text('Plugin example app'),
+        ),
+        body: Center(
+          child: ListView(
+            padding: EdgeInsets.all(5.0),
+            children: <Widget>[
+              Padding(padding: EdgeInsets.all(10.0)),
+              Center(
+                child: RaisedButton(
+                  onPressed: _isAmplifyConfigured ? savePost : null,
+                  child: Text('Save Post'),
+                ),
+              ),
+              Padding(padding: EdgeInsets.all(10.0)),
+              Center(
+                child: RaisedButton(
+                  onPressed: _isAmplifyConfigured ? deletePost : null,
+                  child: Text('Delete Post'),
+                ),
+              ),
+              Padding(padding: EdgeInsets.all(10.0)),
+              Center(
+                child: RaisedButton(
+                  onPressed: _isAmplifyConfigured ? runQueries : null,
+                  child: const Text('Query'),
+                ),
+              ),
+              Padding(padding: EdgeInsets.all(5.0)),
+              // replace with any or all query results as needed
+              Text('Posts >= 4 rating: \n$_posts4rating\n'),
```

Comment:
I think it would be good to keep the other queries for the other data types within the test app here.  Just showing all posts above 4 rating seems quite limiting in terms of data shown.

Reply:
This is just a template. All the variables will have their respective query results, you can choose whatever is relevant to your testing and just update the widget to have that.